### PR TITLE
[feature] support Sec-Websocket-Protocol in streaming API

### PR DESCRIPTION
**Pull request by matthewp**
# Description

Implements `Sec-Websocket-Protocol` header for streaming.

closes https://github.com/superseriousbusiness/gotosocial/issues/1248

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
Have only taken a brief look but this looks really great :) I shall leave it up to @tsmethurst to perform a full review as I am quite limited in my time outside of work :sweat_smile:

File: internal/api/client/streaming/streaming_test.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	testClients      map[string]*gtsmodel.Client
+	testApplications map[string]*gtsmodel.Application
+	testUsers        map[string]*gtsmodel.User
+	testAccounts     map[string]*gtsmodel.Account
+	testAttachments  map[string]*gtsmodel.MediaAttachment
+	testStatuses     map[string]*gtsmodel.Status
+	testFollows      map[string]*gtsmodel.Follow
+
+	// module being tested
+	statusModule    *status.Module
```
Comment (tsmethurst) [inline internal/api/client/streaming/streaming_test.go:73]: can you remove the unused modules?

File: internal/api/client/streaming/streaming.go [hunk 2]
```diff
@@ -35,17 +36,29 @@ const (
 
 	// AccessTokenQueryKey is the query key for an oauth access token that should be passed in streaming requests.
 	AccessTokenQueryKey = "access_token"
+	// AccessTokenHeader is the header for an oauth access token that can be passed in streaming requests instead of
```
Comment (tsmethurst) [inline internal/api/client/streaming/streaming.go:39]: i think you left this comment unfinished :P

File: internal/api/client/streaming/streaming.go [hunk 3]
```diff
@@ -35,17 +36,29 @@ const (
...
+	//nolint:all
```
Comment (tsmethurst) [inline internal/api/client/streaming/streaming.go:40]: better to be specific about which linters you're disabling imo, i assume this is a security related one?

**Review by tsmethurst — COMMENTED** [review]
Looks good, thank you! Couple of comments about a few small things related to tests + comments, but that's it :)

**Comment by matthewp** [comment]
@tsmethurst Updated based on your review

**Comment by tsmethurst** [comment]
great stuff, thank you!

**Pull request merged**